android/engine/src/main/java/org/smartregister/fhircore/engine/util/ApplicationUtil.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.engine.util
+
+import android.app.Application
+
+object ApplicationUtil {
+  /** This Application is used to retrieve string resources outside of the Activity. */
+  lateinit var application: Application
+}
```

Comment:
I do not think we need this class. A utility class just to hold reference to the `Application`.

Reply:
Any suggestions where to put the reference?
I need the reference to be accessible from anywhere.

Reply:
Wait, will this work?
```suggestion
/** This Application is used to retrieve string resources outside of the Activity. */
lateinit var application: Application
```

Reply:
Any reasons for not creating a data class for `SyncStrategy` instead of the limited `enum class` that is currently used?. We can still leverage the multi language support of configurations and avoid this utility function entirely

Reply:
I think Kotlin serializations allows for providing custom serializers which we can use to serialize the `Coding` resource in the  `SyncStrategyConfig` data class

Reply:
@ellykits
Because @dubdabasoduba requested for the tags system and display to be retrieved from strings.xml, it will use the app flavour to configure it.

That's what he meant by configure.

Reply:
@ellykits @FikriMilano Moved the discussion [here](https://github.com/opensrp/fhircore/issues/1657) to allow merging of this PR

Reply:
This is deleted